src/components/GearForm.tsx

```diff
@@ -0,0 +1,307 @@
+    for (let index = 0; index < viewFilters.length; ++index) {
+      if (viewFilters[index].name === tag) {
+        hasMatch = true;
+      }
+    }
+    return hasMatch;
+  }
+
+  function pushViewArray(tag: string): void {
+    const viewArray = viewFilters;
+    const filter = filters.find(function (filter) {
+      return filter.name === tag;
+    });
+    if (filter) {
+      viewArray.push(filter);
+    }
+  }
+
+  function cleanTag(tag: string): string {
+    tag = tag.trim();
+    const words = tag.split(" ");
+    tag = "";
+    for (let index = 0; index < words.length; ++index) {
+      const word = words[index].charAt(0).toUpperCase() + words[index].slice(1);
+      tag = tag + word + " ";
+    }
+    return tag.trim();
+  }
+
+  function filterItems(gear: Gear[], filters: Filter[]): Gear[] | undefined {
+    const tempArray: Gear[] = [];
+    let queriedGear: Gear[] = [];
+    const activeFilters = filters.filter(function (filter) {
+      return filter.toggle;
+    });
+    if (searchString !== "") {
+      const queries = searchString.split(",");
+      for (let i = 0; i < gear.length; ++i) {
+        if (
+          queries.some(function (query) {
+            return (
+              gear[i].title
+                .toLowerCase()
+                .includes(query.toLowerCase().trim()) ||
+              gear[i].tags.toLowerCase().includes(query.toLowerCase().trim())
+            );
+          })
+        ) {
+          queriedGear.push(gear[i]);
+        }
+      }
+    } else {
+      queriedGear = gear;
+    }
+    for (let i = 0; i < queriedGear.length; ++i) {
+      let hasMatch = false;
+      if (matchAny) {
+        if (
+          activeFilters.some(function (filter) {
+            return queriedGear[i].tags
+              .toLowerCase()
+              .includes(filter.name.toLowerCase());
+          })
+        ) {
+          hasMatch = true;
+        }
+      } else {
+        if (
+          activeFilters.every(function (filter) {
+            return queriedGear[i].tags
+              .toLowerCase()
+              .includes(filter.name.toLowerCase());
+          })
+        ) {
+          hasMatch = true;
+        }
+      }
+      if (hasMatch) {
+        tempArray.push(queriedGear[i]);
+      }
+    }
+
+    return tempArray;
+  }
+
+  const toggleFilter = (filter: Filter): void => {
+    filter.toggle = !filter.toggle;
+    const tempArray = [];
+    for (let index = 0; index < filters.length; ++index) {
+      if (filters[index].name == filter.name) {
+        tempArray.push(filter);
+      } else {
+        tempArray.push(filters[index]);
+      }
+    }
+    setFilters(tempArray);
+  };
+
+  function sortFilters(filters: Filter[]): Filter[] {
+    filters.sort(function (a, b) {
+      if (a.name == "") {
+        return -1;
+      } else if (a.name < b.name) {
+        return -1;
+      } else if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+    return filters;
+  }
```

Comment:
This is not the typical way to do an array sort.  Why name the anonymous function and just call it normally?  Why the wrapper?
I could understand a sort wrapper if you wanted to create a **copy** of the array.  But for that you need to call `.slice()` before you sort the array.
```suggestion
    function sortByFilterName (a: Filter, b: Filter) {
      if (if a.name == "" || a.name < b.name) return -1;
      if (a.name > b.name) return 1;
      return 0;
    }
    // usage: filters.sort(sortByFilterName)
```